idTemplate.md

```diff
@@ -0,0 +1,111 @@
+properties of the node (in that dialect instance).
+
+## Declaring an ID template
+ID templates are specified in the *dialect definition*. These are set using the `idTemplate` facet for the node mapping
+responsible for parsing the desired node. The value of an `idTemplate` facet must be a string which can include zero or 
+more ID template variables written between curly braces (e.g. `{myTemplateVariable}`). ID template variables must match 
+one of the property mappings in that same node declared in that same node.
+
+<!-- To debate -->
+Union nodes cannot set ID templates because these don't declare property mappings. ID templates must be set for each
+union member.
+
+## Setting the parsed node ID with an ID template
+<!-- To debate -->
+When parsing a dialect instance, the ID template variables get replaced by the actual **URL-encoded** values (a.k.a. 
+percent-encoded) of the matching property mappings. The resulting string must be a valid URI which will be assigned 
+as that node's ID.
+
+**Note**: The corresponding properties will contain the original non URL-encoded value.
+
+## Valid examples
+### Example 1: simple case
+#### Dialect
+```yaml
+PersonNode:
+  idTemplate: http://people.org/people/{firstName}-{lastName}
+  mapping:
+    firstName:
+      range: string
+      mandatory: true
+    lastName:
+      range: string
+      mandatory: true
+```
+
+#### Dialect Instance
+
+```yaml
+firstName: Lionel
+lastName: Messi
+```
+
+#### Parsed node
+
+```json
+{
+  "@id": "http://people.org/people/Lionel-Messi",
+  "firstName": "Lionel",
+  "lastName": "Messi"
+}
+```
+Notice that the ID of the parsed node gets set using the provided template, replacing the template variables 
+`{firstName}` and `{lastName}` with the values of the parsed property mappings `firstName` and `lastName`, which in this 
+case are `Lionel` and `Messi`
+
+### Example 2: URL-encoded string
+#### Dialect
+```yaml
+PersonNode:
+  idTemplate: http://people.org/people/{fullName}
+  mapping:
+    fullName:
+      range: string
+      mandatory: true
+```
+
+#### Dialect Instance
+
+```yaml
+fullName: Lionel Messi
+```
+
+#### Parsed node
+
+```json
+{
+  "@id": "http://people.org/people/Lionel%20Messi",
+  "fullName": "Lionel Messi"
+}
+```
+Notice that the space character ' ' in the dialect instance gets encoded as `%20` in the node ID but keeps its original
+value in the parsed property.
+
+## Validation of an ID template definition
+When not met the following conditions will raise a **violation** in the dialect definition:
+
+* An ID template variable does not match any property mapping
+<!-- To debate --> 
+<!-- Current behavior: dialect instance definition validates that all variables match some property but also validated 
+that all properties match some property mapping (impossible to define a valid instance!) -->
+
+* An ID template variable matches a non-mandatory property mapping
```

Comment:
Properties should not only be mandatory but marked as primary key as well

Reply:
Property mappings which are of type array should not be included as idTemplate vars